pythonforandroid/build.py

```diff
@@ -709,25 +703,157 @@ def run_pymodules_install(ctx, modules):
+                    output = subprocess.check_output([
+                        "bash", "-c",
+                        "source '" + join(
+                            ctx.build_dir, "venv", "bin", "activate"
+                        ).replace("'", "'\"'\"'") + "' && " +
+                        "pip freeze"
+                    ], env=copy.copy(env))
+                    try:
+                        output = output.encode('utf-8')
+                    except AttributeError:
+                        pass
+                    fileh.write(output)
+
+                info('Populating venv\'s site-packages with '
+                     'ctx.get_site_packages_dir()...')
+
+                # Copy dist contents into site-packages for discovery.
+                # Why this is needed:
+                # --target is somewhat evil and messes with discovery of
+                # packages in PYTHONPATH if that also includes the target
+                # folder. So we need to use the regular virtualenv
+                # site-packages folder instead.
+                # Reference:
+                # https://github.com/pypa/pip/issues/6223
+                ctx_site_packages_dir = os.path.normpath(
+                    os.path.abspath(ctx.get_site_packages_dir())
+                )
+                venv_site_packages_dir = os.path.normpath(os.path.join(
+                    ctx.build_dir, "venv", "lib", [
+                        f for f in os.listdir(os.path.join(
+                            ctx.build_dir, "venv", "lib"
+                        )) if f.startswith("python")
+                    ][0], "site-packages"
+                ))
+                copied_over_contents = []
+                for f in os.listdir(ctx_site_packages_dir):
+                    full_path = os.path.join(ctx_site_packages_dir, f)
+                    if not os.path.exists(os.path.join(
+                                venv_site_packages_dir, f
+                            )):
+                        if os.path.isdir(full_path):
+                            shutil.copytree(full_path, os.path.join(
+                                venv_site_packages_dir, f
+                            ))
+                        else:
+                            shutil.copy2(full_path, os.path.join(
+                                venv_site_packages_dir, f
+                            ))
+                        copied_over_contents.append(f)
+
+                # Get listing of virtualenv's site-packages, to see the
+                # newly added things afterwards & copy them back into
+                # the distribution folder / build context site-packages:
+                previous_venv_contents = os.listdir(venv_site_packages_dir)
+
+                # Actually run setup.py:
+                info('Launch setup.py...')
+                shprint(sh.bash, '-c', (
+                    "source '" + join(
+                        ctx.build_dir, "venv", "bin", "activate"
```

Comment:
Minor: same here, if you're lazy and want to save some lines, you can skip the `source` part and just go:
```python
./venv/bin/pip install -c constraints.txt -v .
````